src/layers/XRLayer/XRLayer.js

```diff
@@ -106,8 +112,11 @@ export default class XRLayer extends Layer {
 
       this.getAttributeManager().invalidateAll();
     }
-    if (props.channelData !== oldProps.channelData) {
-      this.loadTexture(props.channelData);
+    if (
+      props.channelData !== oldProps.channelData &&
+      props.channelData?.data !== oldProps.channelData?.data
+    ) {
+      this.loadChannelTextures(props.channelData);
```

Comment:
This condition was true every time so this resolves that.  No idea why - probably has to do with state or something.